src/scripting/lua_common.hpp

```diff
@@ -207,13 +207,23 @@ int luaW_pcall_internal(lua_State *L, int nArgs, int nRets);
 int luaW_type_error(lua_State *L, int narg, const char *tname);
 int luaW_type_error(lua_State *L, int narg, const char* kpath, const char *tname);
 
+#define deprecate_attrib(name, prefix, level, version, msg) deprecated_message(prefix "." #name, DEP_LEVEL::level, version, msg)
+
+#define return_deprecated_attrib(type_macro, name, accessor, prefix, level, version, msg) \
+	type_macro(name, ( \
+		deprecate_attrib(name, prefix, level, version, msg), \
+		accessor \
+	))
+
 #define return_tstring_attrib(name, accessor) \
 do { \
 	if (strcmp(m, (name)) == 0) { \
 		luaW_pushtstring(L, (accessor)); \
 		return 1; \
 	} \
 } while(false)
+#define return_tstring_attrib_deprecated(name, prefix, level, version, msg, accessor) \
+	return_deprecated_attrib(return_tstring_attrib, name, accessor, prefix, level, version, msg)
```

Comment:
Many of the new `#defines` here for deprecation helpers seem unused - I think we shouldn't add them unless we're going to use them.

Reply:
It seemed easier to add all of them and then they could just be used as needed, rather than having to add each new one as needed.